### DISCUSSION TITLE
Allow for sending a value when deploying a contract

### DISCUSSION THREAD

**Discussion by joaop21** [wevm/viem#542] [Idea / Feature Request] [answered]
This is a possible scenario when deploying a contract. We might want to send some funds on a contract instantiation, and a Solidity contract can be payable.
Not allowing a value to be sent on a contract deployment limits the possible actions we can take here.

**Comment by RielJ** [comment, marked as answer]
I have submitted a PR about this [here](https://github.com/wagmi-dev/viem/pull/822).

**Reply by joaop21** [reply]
Nice! Thx!

**Reply by jxom** [reply]
cheers legend
